test/tbb/test_eh_algorithms.cpp

```diff
@@ -371,7 +371,7 @@ void Test4 () {
         // to do the solitary throw swaps out after registering its intent to throw but before it
         // actually does so. As a result, the number of extra tasks cannot exceed the number of thread
         // for each nested pfor invication)
-        REQUIRE_MESSAGE (g_CurExecuted <= minExecuted + (g_NumThreads-1)*g_NumThreads/2, "Too many tasks survived exception");
+        REQUIRE_MESSAGE (g_CurExecuted <= minExecuted + (g_ExecutedAtLastCatch + g_NumThreads), "Too many tasks survived exception");
```

Comment:
I suggest to remove this assert or rewrite it to: (or refactor code to not duplicate asserts)
```suggestion
REQUIRE_MESSAGE (g_CurExecuted <= g_ExecutedAtLastCatch + g_NumThreads, "Too many tasks survived exception");
```

Reply:
`minExecuted` is required because only one exception is thrown in this branch, i.e. only one nested loop is canceled while other nested loops will be fully executed.